lib/src/tabs/Tab.tsx

```diff
@@ -0,0 +1,182 @@
+  display: inline-flex;
+  align-items: center;
+  user-select: none;
+  vertical-align: middle;
+  justify-content: center;
+  opacity: 0.7;
+  border-bottom-style: solid;
+  border-bottom-color: ${(props) => props.theme.dividerColor};
+  font-family: ${(props) => props.theme.fontFamily};
+  font-size: ${(props) => props.theme.fontSize};
+  font-style: ${(props) => props.theme.fontStyle};
+  font-weight: ${(props) => props.theme.fontWeight};
+  min-width: 90px;
+  max-width: 360px;
+  padding: ${(props) =>
+    ((!props.hasLabelAndIcon || (props.hasLabelAndIcon && props.iconPosition !== "top")) && "12px 16px") || "8px 16px"};
+  height: ${(props) =>
+    ((!props.hasLabelAndIcon || (props.hasLabelAndIcon && props.iconPosition !== "top")) && "48px") || "72px"};
+  min-height: ${(props) =>
+    ((!props.hasLabelAndIcon || (props.hasLabelAndIcon && props.iconPosition !== "top")) && "48px") || "72px"};
+  background-color: ${(props) => props.theme.unselectedBackgroundColor};
+  color: ${(props) => props.theme.unselectedFontColor};
+  svg {
+    color: ${(props) => props.theme.unselectedIconColor};
+  }
+
+  &:hover {
+    background-color: ${(props) => `${props.theme.hoverBackgroundColor} !important`};
+  }
+
+  &:active {
+    background-color: ${(props) => `${props.theme.pressedBackgroundColor} !important`};
+    font-weight: ${(props) => `${props.theme.pressedFontWeight} !important`};
+  }
+
+  &:focus {
+    outline: ${(props) => props.theme.focusOutline} auto 1px;
+  }
+
+  &[aria-selected="true"] {
+    background-color: ${(props) => props.theme.selectedBackgroundColor};
+    color: ${(props) => props.theme.selectedFontColor};
+    svg {
+      color: ${(props) => props.theme.selectedIconColor};
+    }
+    opacity: 1;
+  }
+
+  &[aria-disabled="true"] {
+    background-color: ${(props) => props.theme.unselectedBackgroundColor} !important;
+    cursor: not-allowed !important;
+    pointer-events: all;
+    color: ${(props) => props.theme.disabledFontColor};
+    font-style: ${(props) => props.theme.disabledFontStyle};
+    svg {
+      color: ${(props) => props.theme.disabledIconColor};
+    }
+    outline: none !important;
+    opacity: 0.5;
+  }
+`;
+
+const ParentLabelSpan = styled.div`
+  position: relative;
+`;
+
+const TabLabelContainer = styled.div<IconProps>`
+  display: flex;
+  flex-direction: ${(props) => (props.hasLabelAndIcon && props.iconPosition === "top" && "column") || "row"};
+  align-items: center;
+`;
+
+const LabelTextContainer = styled.div``;
+
+const BadgeContainer = styled.div<IconProps>`
+  position: absolute;
+  right: 0;
+  top: ${(props) => (props.hasLabelAndIcon && props.iconPosition === "top" && "0") || "5px"};
+  width: 23px;
+  height: 17px;
+`;
+
+const MainLabelContainer = styled.div<{ hasBadge: number | boolean }>`
+  display: flex;
+  flex-direction: row;
+  margin-left: ${(props) => (props.hasBadge && "35px") || "unset"};
+  margin-right: ${(props) => (props.hasBadge && "35px") || "unset"};
+`;
```

Comment:
That´s true, I will take a look.

Reply:
Using ref to take the width of the badge didn't work as expected because of the re-renders and the selected underline. So I used the calculation: badge width + the margin left (margin-left=12px and width of the badge is 16px or 23px depending on the notification number).